PyInstaller/building/makespec.py

```diff
@@ -759,10 +701,7 @@ def main(
     if is_win and os.path.splitext(scripts[0])[-1] == '.pyw':
         console = False
 
-    # If script paths are relative, make them relative to the directory containing .spec file.
-    scripts = [make_path_spec_relative(x, specpath) for x in scripts]
-    # With absolute paths replace prefix with variable HOMEPATH.
-    scripts = list(map(Path, scripts))
+    scripts = list(map(portable_filepath, scripts))
```

Comment:
I think these script paths still need to be made spec-relative, though, because the build process assumes that if they are relative, they are relative to spec directory.

For example, this used to work

`pyinstaller --clean --noconfirm program/program.py --specpath spec/`

and now it does not (and in contrast to the `HOMEPATH` thing, I imagine this is used). Although I suspect that this scenario is party broken anyway, because *all* given relative paths (icons, etc.) that we later treat as spec-relative would need to be made relative to spec here... But for now, we should at least maintain status quo here.